Remplacement de settings.py dans la documentation (#6707)

Convolutio opened the pull request:
Un fichier `settings.py` est mentionné dans la documentation pour contenir des éléments de configuration tandis qu'ils sont situés dans le fichier `zds/settings/abstract_base/django.py`

`settings.py` a été temporairement laissé pour les champs de configuration de la pagination, dont le véritable emplacement n'a pas été trouvé.

<!-- Décrivez vos changements ici (optionnel pour les tous petits changements). -->

<!-- Remplacez XXXX par le numéro de ticket corrigé par vos changements : GitHub fermera le ticket mentionné automatiquement (voir https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue pour plus de détails). Supprimez la ligne s'il n'y a pas de ticket associé. --!>
Fix #6707 

<!-- Si votre PR modifie du code Python, mettez à jour ou créez les tests unitaires associés. Signalez vos éventuelles difficultés afin que des contributeurs expérimentés puissent vous aider. -->

<!-- Si votre pull request requiert des actions particulières lors de la mise en production, renseignez-les ici afin qu’elles soient ajoutées au changelog lors du merge. -->

### Contrôle qualité

<!-- Donnez des instructions pour nous aider à vérifier vos changements.

Par exemple :

  - Lancez `python manage.py migrate` et `yarn test` ;
  - Créez un nouveau compte nommé `toto` ;
  - Envoyez un message privé à quelqu’un d’autre, le titre du message doit apparaître en rose. -->

<!-- Merci ! -->

- Recharger la documentation avec `make generate-doc`
- Servir la page de documentation avec `python3 -m http.server 8080 -d doc/build/html/`
- Observer les changements depuis le navigateur à la page `localhost:8080/api.html`

Convolutio committed:
Remplacement de settings.py dans la documentation (#6707)

Un fichier `settings.py` est mentionné dans la documentation
pour contenir des éléments de configuration tandis qu'ils sont situés
dans le fichier `zds/settings/abstract_base/django.py`

`settings.py` a été laissé pour les champs de configuration de
la pagination, dont le véritable emplacement n'a pas été trouvé.

philippemilink commented:
> settings.py a été temporairement laissé pour les champs de configuration de la pagination, dont le véritable emplacement n'a pas été trouvé.

La partie de la doc concernant la pagination ne semble pas du tout à jour, la dernière mention de `PAGINATE_BY_PARAM` est dans le commit addf026520b586890a5696c651fccc2166e23c9e (qui date de 2016 !).

Il faudrait donc en profiter pour mettre cette partie de la doc à jour. J'ai l'impression qu'il suffit de reprendre ce qui est mentionné dans la doc officielle de DjangoRestFramework concernant la pagination (le lien est déjà dans la doc de ZdS) et dire que c'est comme ça qu'on l'utilise dans ZdS.

philippemilink commented:
> Il faudrait donc en profiter pour mettre cette partie de la doc à jour. J'ai l'impression qu'il suffit de reprendre ce qui est mentionné dans la doc officielle de DjangoRestFramework concernant la pagination (le lien est déjà dans la doc de ZdS) et dire que c'est comme ça qu'on l'utilise dans ZdS.

Je propose la modification suivante : 
```diff
diff --git a/doc/source/api.rst b/doc/source/api.rst
index 8a87dc51c..9d5293a5b 100644
--- a/doc/source/api.rst
+++ b/doc/source/api.rst
@@ -171,14 +171,12 @@ Pagination
 
 La pagination permet d'éviter au serveur de faire des requêtes trop lourdes sur la base de données. Par exemple, si un client désire récupérer la liste de tous les utilisateurs de la plateforme et que cette même plateforme dispose d'un très grand nombre d'utilisateurs, la requête en base de données pourrait être lourde. Coupler à ceci des intentions malveillantes pour faire tomber le serveur, paginer les listes de ressources est presque une mesure de sécurité.
 
-La pagination peut être configurée directement dans les vues de l'API mais aussi dans le fichier ``settings.py`` pour s'appliquer à l'ensemble des listes des ressources de l'API. Dans le fichier ``settings.py``, ``PAGINATE_BY`` renseigne la taille d'une page, ``PAGINATE_BY_PARAM`` permet aux clients de modifier la taille d'une page et ``MAX_PAGINATE_BY`` permet d'imposer une taille maximale.
+La pagination peut être configurée directement dans les vues de l'API mais aussi par le biais d'une classe Python. Cette classe doit être renseignée dans le configuration de l'API dans le fichier ``zds/settings/abstract_base/django.py``.
 
 .. sourcecode:: python
 
     REST_FRAMEWORK = {
-        'PAGINATE_BY': 10,                  # Default to 10
-        'PAGINATE_BY_PARAM': 'page_size',   # Allow client to override, using `?page_size=xxx`.
-        'MAX_PAGINATE_BY': 100,             # Maximum limit allowed when using `?page_size=xxx`.
+        "DEFAULT_PAGINATION_CLASS": "zds.api.pagination.DefaultPagination",
     }
 
 Toutes les informations complémentaires à ce sujet sont disponibles dans la `documentation de la pagination (en) <http://www.django-rest-framework.org/api-guide/pagination/>`_.
```

philippemilink commented:
En fait, il faudrait en profiter pour corriger toutes les références à `settings.py` dans la documentation, pas que dans la partie concernant l'API. Je propose les modifications en PJ : [remove-settings-doc.diff.txt](https://github.com/user-attachments/files/19376561/remove-settings-doc.diff.txt) (GitHub refuse d'uploader un `.diff`, mais c'est bien un diff, il suffit de `git apply < remove-settings-doc.diff.txt` pour l'appliquer).

Convolutio committed:
Mise à jour de la documentation de la pagination

Mise à jour des mentions des paramètres de REST_FRAMEWORK
selon ceux des dernières version de Django.

Correction des références à `settings.py` dans la documentation

philippemilink committed:
Remplace un dernier settings.py oublié dans la doc

philippemilink reviewed (approved):
En greppant dans le code source de la doc, j'ai vu encore un dernier `settings.py` qui traînait, je l'ai corrigé.

**QA OK** :heavy_check_mark:
